src/Sarif.WorkItems/SarifWorkItemsExtensions.cs

```diff
@@ -103,20 +104,47 @@ private static string ConstructFullRuleIdentifier(ReportingDescriptor reportingD
             return fullRuleIdentifier;
         }
 
-        public static string CreateWorkItemDescription(this SarifLog log)
+        public static string CreateWorkItemDescription(this SarifLog log, Uri locationUri)
         {
-            Dictionary<string, int> resultCountsByTool = ComputeToolResultCounts(log);
-            StringBuilder templateText = new StringBuilder(@"This bug contains results for the below tool(s) and issue(s).  It was filed automatically.");
-            templateText.AppendLine();
-            templateText.AppendLine();
+            Dictionary<Run, int> resultCountsByRun = ComputeRunResultCounts(log);
+            StringBuilder templateText = new StringBuilder();
+            int runningResults = 0;
+            string toolNames = string.Empty;
+            string multipleToolsFooter = string.Empty;
+
+            Uri runRepositoryUri = resultCountsByRun.FirstOrDefault().Key?.VersionControlProvenance?.FirstOrDefault().RepositoryUri;
+            string detectionLocation = !string.IsNullOrEmpty(runRepositoryUri?.OriginalString) ? runRepositoryUri?.OriginalString : locationUri?.OriginalString;
+
+            toolNames = string.Format("'{0}'", resultCountsByRun.FirstOrDefault().Key?.Tool?.Driver?.Name);
+            runningResults = resultCountsByRun.FirstOrDefault().Value;
 
-            foreach (KeyValuePair<string, int> toolResult in resultCountsByTool)
+            if (resultCountsByRun.Count > 1)
             {
-                templateText.Append(string.Format("*Tool: {0}", toolResult.Key));
-                templateText.AppendLine();
-                templateText.Append(string.Format("     Result count: {0}", toolResult.Value));
-                templateText.AppendLine();
+                Run lastrun = resultCountsByRun.Last().Key;
+                multipleToolsFooter = " (and other locations)";
+                foreach (KeyValuePair<Run, int> run in resultCountsByRun)
```

Comment:
>run [](start = 48, length = 3)

Let's use a different variable name since `run` would usually be, well, a `Run`. I suggest `pair`, or I sometimes use `entry` (as in "dictionary entry").